### ISSUE TITLE
error connecting to hub

### ISSUE THREAD

**Issue by darkoddio** [lopelex/node-red-contrib-harmony#2]
Hi, I'm tying to use the HWS nodes, but since my Harmony Hub is not always turned on, I get several messages like the one below: I was using just the "HWS activity" to fire some activity, and I do not understand why the node (or the configuration node) should connect to the hub if I'm not requesting to lunch any activity.
Most important thing is that I'm getting the "Error: connect EHOSTUNREACH 192.168.1.128:8088" message even if I have removed ANY node related to HWS from my flows, including the configuration one.

`{ RequestError: Error: connect EHOSTUNREACH 192.168.1.128:8088
    at new RequestError (/data/node_modules/harmony-websocket/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/data/node_modules/harmony-websocket/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/data/node_modules/harmony-websocket/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/data/node_modules/harmony-websocket/node_modules/request/request.js:185:22)
    at emitOne (events.js:115:13)
    at Request.emit (events.js:210:7)
    at Request.onRequestError (/data/node_modules/harmony-websocket/node_modules/request/request.js:881:8)
    at emitOne (events.js:120:20)
    at ClientRequest.emit (events.js:210:7)
    at Socket.socketErrorListener (_http_client.js:400:9)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at emitErrorNT (internal/streams/destroy.js:62:8)
    at _combinedTickCallback (internal/process/next_tick.js:102:11)
    at process._tickCallback (internal/process/next_tick.js:161:9)
  name: 'RequestError',
  message: 'Error: connect EHOSTUNREACH 192.168.1.128:8088',
  cause:
   { Error: connect EHOSTUNREACH 192.168.1.128:8088
       at Object.exports._errnoException (util.js:1022:11)
       at exports._exceptionWithHostPort (util.js:1045:20)
       at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:14)
     code: 'EHOSTUNREACH',
     errno: 'EHOSTUNREACH',
     syscall: 'connect',
     address: '192.168.1.128',
     port: 8088 },
  error:
   { Error: connect EHOSTUNREACH 192.168.1.128:8088
       at Object.exports._errnoException (util.js:1022:11)
       at exports._exceptionWithHostPort (util.js:1045:20)
       at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:14)
     code: 'EHOSTUNREACH',
     errno: 'EHOSTUNREACH',
     syscall: 'connect',
     address: '192.168.1.128',
     port: 8088 },
  options:
   { url: 'http://192.168.1.128:8088/',
     method: 'POST',
     timeout: 10000,
     headers:
      { Origin: 'http://sl.dhg.myharmony.com',
        'Content-Type': 'application/json',
        Accept: 'application/json',
        'Accept-Charset': 'utf-8' },
     json: true,
     body: { id: 1, cmd: 'setup.account?getProvisionInfo', params: {} },
     callback: [Function: RP$callback],
     transform: undefined,
     simple: true,
     resolveWithFullResponse: false,
     transform2xxOnly: false },
  response: undefined }
Error: connect EHOSTUNREACH 192.168.1.128:8088
`

**Comment by lopelex** [comment]
ready for testing
https://github.com/lopelex/node-red-contrib-harmony/commit/1bd0a1bb3d74c52c6cdbe1a3cb3cce0ce001e26e
npm install https://github.com/lopelex/node-red-contrib-harmony.git

**Comment by ristomatti** [comment]
I've ran across this also even though I have the Hub always powered (likely a network connectivity issue). Is there a periodical connection retry implemented or is a new connection always opened for each request?

I'm asking as I have a faint memory that after unplugging the Hub, I've had to restart Node-RED flows for it to reconnect. I might mix it to some other hub/smart device (and it's Node-RED plugin) I have though.
